python/GangaDirac/Lib/Utilities/DiracUtilities.py

```diff
@@ -237,19 +175,17 @@ def execute(command,
         python_setup (str): Optional extra code to pass to python when executing
         eval_includes (???): TODO document me
         update_env (bool): Should this modify the given env object with the env after the command has executed
+        cred_req (ICredentialRequirement): What credentials does this call need
     """
 
     if env is None:
         env = getDiracEnv()
     if python_setup == '':
         python_setup = getDiracCommandIncludes()
 
-    # We're about to perform an expensive operation so being safe before we run it shouldn't cost too much
-    _checkProxy(force = True)
-
-    #logger.debug("Executing command:\n'%s'" % str(command))
-    #logger.debug("python_setup:\n'%s'" % str(python_setup))
-    #logger.debug("eval_includes:\n'%s'" % str(eval_includes))
+    if cred_req is not None:
+        env = env or {}
+        env['X509_USER_PROXY'] = credential_store[cred_req].location
```

Comment:
Ahh, this is a change from when this code was first written. Now we have
```python
if env is None:
    env = getDiracEnv()
```
I can get rid of this so that it's:
```python
if cred_req is not None:
    env['X509_USER_PROXY'] = credential_store[cred_req].location
```